pkg/bootstrap/crds.go

```diff
@@ -129,6 +130,7 @@ func processCRDFile(ctx context.Context, cl client.Client, fileName string) erro
 			if errors.IsNotFound(err) {
 				log.Info("creating CRD", "file", fileName, "index", index, "CRD", obj.GetName())
 				err = cl.Create(ctx, obj)
+				err = hacks.WorkAroundTypeObjectProblemInCRDSchemas(ctx, err, cl, obj)
```

Comment:
Could we be proactive, by reading the underlying k8s version and doing the right thing?

Reply:
Yeah, not sure about that due to a few reasons:
- how do we know which (future) version of OpenShift will handle this properly (OpenShift could be patched specifically to fix this problem, so the underlying k8s version might not matter)
- theoretically, the API server can be upgraded without restarting the operator; we couldn't just check the version at operator startup
- it may also be possible that the operator hits a different version of the API server on every request (e.g. during an upgrade of an H/A installation of the API Server, some instances may run an older version, while others run a new one; the load balancer in front of the API server thus routes the operator's requests to different API server versions).

Reply:
Understood.  Maybe we could set a flag on the first error, so we don't have to do the error processing on every CRD?

Reply:
I think it's fine as it is - we will only do the comparison if `cl.Create()` returned an error, which should be rare enough